arangod/Graph/PathManagement/PathValidator.cpp

```diff
@@ -302,6 +302,22 @@ auto PathValidator<ProviderType, PathStore, vertexUniqueness, edgeUniqueness>::
     }
   }
 
+  auto edgeExpr = _options.getEdgeExpression();
+  if (edgeExpr != nullptr) {
+    if (step.getEdge().isValid()) {
+      edgeBuilder.clear();
+
+      _provider.addEdgeToBuilder(step.getEdge(), edgeBuilder);
+      bool satisfiesCondition =
+          evaluateExpression(edgeExpr, edgeBuilder.slice());
+      if (!satisfiesCondition) {
+        return ValidationResult{ValidationResult::Type::FILTER_AND_PRUNE};
+      }
+    } else {
+      // TODO: at the moment we smile and wave...
+    }
```

Comment:
Would just remove that block - as it does nothing right now.

Reply:
I put that comment in there after debugging tests to make it clear that invalid edges just pass this check.

Tbh I don't entirely know what an "invalid" edge is and under which circumstances they would end up here. If I ADB_PROD_ASSERT() here, I think everything explodes